transform_assumption.ml

```diff
@@ -58,22 +58,33 @@ module TargetSet = Set.Make(LabelTarget)
 
 (* Removes all empty checkpoints with no incoming references *)
 let remove_empty_osr : opt_prog = fun prog ->
-  let get_targets = function[@warning "-4"]
-    | Osr {target; cond; frame_maps} when cond <> [] ->
-      let add tgs {cont_pos} = TargetSet.union tgs (TargetSet.singleton cont_pos) in
-      List.fold_left add (TargetSet.singleton target) frame_maps
+  let get_targets func version seen = function[@warning "-4"]
+    | Osr {label; target; cond; frame_maps} ->
+      let my_pos = { func = func.name; version = version.label; pos = label } in
+      if cond <> [] || TargetSet.mem my_pos seen then begin
+        let add tgs {cont_pos} = TargetSet.union tgs (TargetSet.singleton cont_pos) in
```

Comment:
Right, but that only justifies adding the *assumption* target to the seen-set. My question was about the `cont_pos`, the labels that are not jumped to on deoptimization, but describe extra stack frames.